### ISSUE TITLE
Question: Updating columns defined as expressions

### ISSUE THREAD

**Issue by gregg1ep00** [volkanceylan/Serenity#92]
Suppose I have two tables:

```
CREATE TABLE People (
  PersonId int IDENTITY(1,1) NOT NULL,
  Name nvarchar(250) NOT NULL,
  SchoolId int NOT NULL,
  CONSTRAINT PK_People PRIMARY KEY CLUSTERED (
    PersonId ASC
  )
)

CREATE TABLE Schools (
  SchoolId int IDENTITY(1,1) NOT NULL,
  Name nvarchar(250) NOT NULL,
  CONSTRAINT Schools PRIMARY KEY CLUSTERED (
    SchoolId ASC
  )
)

ALTER TABLE People 
  WITH CHECK ADD CONSTRAINT FK_People_Schools 
    FOREIGN KEY SchoolId
    REFERENCES Schools (SchoolId)

ALTER TABLE People
  CHECK CONSTRAINT FK_People_Schools
```

Suppose I also have PersonRow and SchoolRow defined as follows:
```
public class PersonRow : Row, IIdRow, INameRow
{
  [DisplayName("Id"), Identity]
  public Int32? PersonId
  {
    get { return Fields.PersonId[this]; }
    set { Fields.PersonId[this] = value; }
  }

  [DisplayName("Name"), NotNull, QuickSearch]
  public String Name
  {
    get { return Fields.Name[this]; }
    set { Fields.Name[this] = value; }
  }

  [DisplayName("School"), NotNull, ForeignKey("Schools", "SchoolId"), LeftJoin("jSchool")]
  public Int32? SchoolId
  {
    get { return Fields.SchoolId[this]; }
    set { Fields.SchoolId[this] = value; }
  }

  [DisplayName("School Name"), Expression("jSchool.Name"), QuickSearch]
  public String SchoolName
  {
    get { return Fields.SchoolName[this]; }
    set { Fields.SchoolName[this] = value; }
  }
}


public class SchoolRow : Row, IIdRow, INameRow
{
  [DisplayName("Id"), Identity]
  public Int32? SchoolId
  {
    get { return Fields.SchoolId[this]; }
    set { Fields.SchoolId[this] = value; }
  }

  [DisplayName("Name", NotNull)]
  public String Name
  {
    get { return Fields.Name[this]; }
    set { Fields.Name[this] = value; }
  }
}
```

Can I show a single dialog that allows me to EDIT both the name of the person and the name of the school in a single operation?  I don't want to select the school here, I want to actually update the data of the School row as well as update the data of the Person row.

I realize this is a bit of a contrived example, but it illustrates what I'd like to accomplish.

Thanks!

**Comment by zehranur** [comment]
maybe you can do like that; 
add a new property in row which has SetFieldFlags(FieldFlags.ClientSide) attribute 
    [DisplayName("My Column"), Size(50), SetFieldFlags(FieldFlags.ClientSide)]
        public String MyColumn
        {
            get { return Fields.MyColumn[this]; }
            set { Fields.MyColumn[this] = value; }
        }

and can set any data in ur dialog .
then override BeforeSave in repository 
then set that client side column which row u want to change

 protected override void BeforeSave()
            {
base.BeforeSave();
Row.WhichIsIWannaChange = Row.MyColumn;
.....
...
}

in short, we can add a ClientSide property in our row then in repository we can set it which property we want to add.

**Comment by gregg1ep00** [comment]
Let me elaborate a bit on the data, just so it's more clear of what I'm trying to accomplish.

The Person object has some data that I only need to retain the latest information on (for example, Name, Address, Gender, etc.).  Some aspects of a Person object, however, may change annually, and the system should track a history of this (i.e., Height, Weight, FavoriteColor).

So maybe this is a better example:
```
public class PersonRow : Row, IIdRow, INameRow
{
    [DisplayName("Id"), Identity]
    public int PersonId
    {
        get { return Fields.PersonId[this]; }
        set { Fields.PersonId[this] = value; }
    }

    [DisplayName("Name"), NotNull, QuickSearch]
    public string Name
    {
        get { return Fields.Name[this]; }
        set { Fields.Name[this] = value; }
    }

    [DisplayName("Address"), QuickSearch]
    public string Address
    {
        get { return Fields.Address[this]; }
        set { Fields.Address[this] = value; }
    }
}


public class PersonExtDataRow : Row, IIdRow, INameRow
{
    [DisplayName("Id"), Identity]
    public int EntryId
    {
        get { return Fields.EntryId[this]; }
        set { Fields.EntryId[this] = value; }
    }

    public DateTime Timestamp
    {
        get { return Fields.Timestamp[this]; }
        set { Fields.Timestamp[this] = value; }
    }

    [DisplayName("Person"), NotNull, ForeignKey("People", "PersonId"), LeftJoin("p")]
    public int PersonId
    {
        get { return Fields.PersonId[this]; }
        set { Fields.PersonId[this] = value; }
    }

    [DisplayName("Name"), Expression("p.Name"), QuickSearch]
    public string PersonName
    {
        get { return Fields.PersonName[this]; }
        set { Fields.PersonName[this] = value; }
    }

    [DisplayName("Address"), Expression("p.Address")]
    public string PersonAddress
    {
        get { return Fields.PersonAddress[this]; }
        set { Fields.PersonAddress[this] = value; }
    }

    [DisplayName("Height")]
    public string Height
    {
        get { return Fields.Height[this]; }
        set { Fields.Height[this] = value; }
    }

    [DisplayName("Weight")]
    public string Weight
    {
        get { return Fields.Weight[this]; }
        set { Fields.Weight[this] = value; }
    }

    [DisplayName("FavoriteColor")]
    public string FavoriteColor
    {
        get { return Fields.FavoriteColor[this]; }
        set { Fields.FavoriteColor[this] = value; }
    }
}
```

So the idea is that periodically, the Person would be asked to update his profile, which would consist of the union of the Person table and the PersonExtData row.  If the person's Address changes, I want that to be a global change.  But if the FavoriteColor changes, that should be a tracked change with full history.  Every time the Person changes his profile, a new row would be added to PersonExtData.

Does that make more sense?

**Comment by volkanceylan** [comment]
What Zehranur above said is close to solution, maybe just not clear. You can override AfterSave method of update handler for Person, and can update/insert another table simply. Of course, it is better to check if there are changes in fields related to log table.

Anyway, what you are asking is very similar to CaptureLog system integrated to Serenity. See Northwind ProductRow and ProductLogRow tables. ProductLog is a capture log (history) table of Product records. It keeps changes, plus also allows temporal queries, like how was my product tree looking at X point of time. 

You may put any fields you want to be capture logged in ProductLog. Don't have to log them all. I think this would be a better design for your needs too.

**Comment by gregg1ep00** [comment]
Awesome, thanks so much guys.  Really appreciate the help.  Great work!
